presto-spark-base/src/main/java/com/facebook/presto/spark/execution/AbstractPrestoSparkQueryExecution.java

```diff
@@ -564,6 +573,56 @@ protected void validateStorageCapabilities(TempStorage tempStorage)
         }
     }
 
+    private void collectMetrics(TaskInfo taskInfo)
+    {
+        int taskId = taskInfo.getTaskId().getId();
+        int stageId = taskInfo.getTaskId().getStageExecutionId().getStageId().getId();
+        List<Map<String, Long>> newStatsList = new ArrayList<>();
```

Comment:
There is a list for each Task. Check collectMetrics(taskInfo) in line 635 above

Reply:
@AdeleoM I'm not sure I understand. Isn't there only 1 task? 

CC: @miaoever

Reply:
Isn't each taskInfo representing a single task?

Reply:
We are getting more taskInfo because we get the data on the query completion. 
https://github.com/prestodb/presto/blob/master/presto-spark-base/src/main/java/com/facebook/presto/spark/execution/AbstractPrestoSparkQueryExecution.java#L575

Reply:
Not sure I understand. Does this code run on the executor or driver node? Or, does it run on both?

Reply:
@miaoever  can you please clarify?  I believe Masha has some concern over pulling task stats at this level.

Reply:
@miaoever @AdeleoM MJ, Adele, would you share more details about the new plan?

Reply:
> @miaoever @AdeleoM MJ, Adele, would you share more details about the new plan?

@mbasmanova as my discussion with @AdeleoM, I think we should move the metrics deriving from here (on driver) to PrestoSparkTaskExecutorFactory (on executor) in order to get the task level metrics update in Spark.

Reply:
@miaoever Just to clarify, execution/AbstractPrestoSparkQueryExecution.java runs on the driver, but PrestoSparkTaskExecutorFactory runs on executor and we will be moving this logic into PrestoSparkTaskExecutorFactory.

Reply:
> @miaoever Just to clarify, execution/AbstractPrestoSparkQueryExecution.java runs on the driver, but PrestoSparkTaskExecutorFactory runs on executor and we will be moving this logic into PrestoSparkTaskExecutorFactory.

Exactly.